fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1676,18 +1692,25 @@ def _get_default_cuda_device(module: nn.Module) -> torch.device:
     return torch.device("cuda")
 
 
-@torch.no_grad()
-def cast_inputs_to_fp16(*args: Any, **kwargs: Any) -> Tuple[Any, Any]:
+def cast_inputs_to_dtype(to_fp16: bool, no_grad: bool, *args: Any, **kwargs: Any) -> Tuple[Any, Any]:
```

Comment:
nit, but it feels a bit confusing, cast_to_dtype is a generic term, but in essence you get a fp16 bool behind that in the interface

Reply:
Good point. Let me change to "cast_floats_to_right_precision".